### PR TITLE
Initial jsdoc commit

### PR DESCRIPTION
**Pull request by StefanEnberg**

### CODE REVIEW

File: src/connection-matcher.js [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+   */
   constructor() {
     this.connectionsRegistry = [];
   }
 
+  /**
+   * Add a matching function with a connection instance
+   * @private
+   * @param matchingFn - Matching function to decide what connection function to invoke
+   * @param connectionFn - Should return a connection instance
```
Comment (wennmo) [inline src/connection-matcher.js]: `Should`? Guess it will return a connection instance otherwise an error?

File: src/connection-matcher.js [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
     this.connectionsRegistry.push({
       matchingFn,
       connection,
     });
   }
 
+  /**
+   * Find a match for connection based on the input data
+   * @private
+   * @param {string} data - Url, file path, csv data
```
Comment (wennmo) [inline src/connection-matcher.js]: Are these examples of params? Perhaps elaborate on what `data` can be?

File: src/connections/connection-base.js [hunk 3]
```diff
@@ -1,20 +1,41 @@
 import { uniqueName } from '../utils/utils';
 
 class ConnectionBase {
+  /**
+   * A base set of methods used by Connections
+   * @public
+   * @param {string} path - The Path to a resource
+   * @param {string} connectionType
```
Comment (wennmo) [inline src/connections/connection-base.js]: Is there a definition of what a `connectionType` can be?

File: src/halyard.js [hunk 4]
```diff
@@ -63,16 +96,32 @@ class Halyard {
...
     return this.getAllScriptBlocks().map(item => this.getItemScript(item))
       .join(SCRIPT_BLOCK_SPACING);
   }
 
-  // Support to add hyper cube explicit or implicitly
+  /**
+   * Add hyper cube explicit or implicitly
+   * @public
+   * @param {object} arg1 - Hypercube
+   * @param {object} options
```
Comment (wennmo) [inline src/halyard.js]: what are the `options`?

File: src/halyard.js [hunk 5]
```diff
@@ -93,7 +142,13 @@ class Halyard {
     return newHyperCube;
   }
 
-  // Support to add table explicit or implicitly
+  /**
+   * Support to add table explicit or implicitly
+   * @public
+   * @param {object} arg1 - Table
+   * @param options
```
Comment (wennmo) [inline src/halyard.js]: What are the `options`?

File: src/halyard.js [hunk 6]
```diff
@@ -114,6 +174,12 @@ class Halyard {
     }
   }
 
+  /**
+   * Add new item to the model
```
Comment (wennmo) [inline src/halyard.js]: If an item with that name already exists, should we still return `newItem`? Are we sure it's the same item only based on the name?

File: src/hyper-cube.js [hunk 7]
```diff
@@ -174,6 +240,11 @@ class HyperCube {
     return data;
   }
 
+  /**
+   * Get the fields from the hyper cube
+   * @private
+   * @returns {object[]}
```
Comment (wennmo) [inline src/hyper-cube.js]: Is it an array of field strings that are returned?

File: src/set-statement.js [hunk 8]
```diff
@@ -1,15 +1,31 @@
...
+   */
   getScript() {
     return Object.keys(this.defaultSetStatements)
       .map(key => `SET ${key}='${Array.isArray(this.defaultSetStatements[key])
         ? this.defaultSetStatements[key].join(';') : this.defaultSetStatements[key]}';`)
       .join('\n');
   }
 
+  /**
+   * Returns the name but since statement doesn't have names it will be empty
```
Comment (wennmo) [inline src/set-statement.js]: Will this always return empty? The description kind of implies that :)

File: src/table.js [hunk 9]
```diff
@@ -62,17 +83,32 @@ class Table {
...
+   * Is the table used as a proceeding load
+   * @public
+   * @returns {boolean}
+   */
   isProceedingLoad() {
     return this.connection instanceof Table;
   }
 
+  /**
+   * Get specified prefix
```
Comment (wennmo) [inline src/table.js]: Perhaps elaborate more

File: src/utils/hyper-cube-utils.js [hunk 10]
```diff
@@ -56,6 +92,12 @@ function isDimensionTypeInterval(dimension) {
   return false;
 }
 
+/**
+ * Get dimension type
+ * @private
+ * @param {object} dimension
+ * @returns {string}
```
Comment (wennmo) [inline src/utils/hyper-cube-utils.js]: What types are there?

File: src/utils/hyper-cube-utils.js [hunk 11]
```diff
@@ -88,6 +137,13 @@ function isNumericDimensionType(dimensionType) {
...
   return numericDimensionTypes.indexOf(dimensionType.toLowerCase()) > -1;
 }
+
+/**
+ * Is field numeric
+ * @private
+ * @param {object} field
+ * @returns {boolean}
+ */
 function storeNumeric(field) {
```
Comment (wennmo) [inline src/utils/hyper-cube-utils.js]: Why is it called `storeNumeric`? Does it do more than checking if a field is numeric?

File: src/hyper-cube.js [hunk 12]
```diff
@@ -174,6 +240,11 @@ class HyperCube {
     return data;
   }
 
+  /**
+   * Get the fields from the hyper cube
```
Comment (FredrikFolkesson) [inline src/hyper-cube.js]: hypercube

File: src/derived-fields-template.js [hunk 13]
```diff
@@ -1,13 +1,28 @@
 import { getFieldName, escapeText } from './utils/utils';
 
 class DerivedFieldsTemplate {
+  /**
+   * Declare fields that can be derived from a template. An example can be a calendar template.
+   * @public
+   * @param {object} options - Derived Field Template definition.
```
[the remark this answers is not in the document]
Comment (StefanEnberg) [reply src/derived-fields-template.js]: They are on the lines below 7 :)

File: src/calendar-derived-fields.js [hunk 14]
```diff
@@ -22,6 +22,12 @@ Month(Today())-Month($1) AS [MonthRelNo] ,
 (WeekStart(Today())-WeekStart($1))/7 AS [WeeksAgo] ,
 Week(Today())-Week($1) AS [WeekRelNo];`;
 
+/**
+ * Get the derived field definition for a field that matches the pattern
+ * @public
+ * @param fn - Field matcher function
```
Comment (miralemd) [inline src/calendar-derived-fields.js]: missing type on `fn`

File: src/connection-matcher.js [hunk 15]
```diff
@@ -1,15 +1,32 @@
...
+   * @constructor
+   */
   constructor() {
     this.connectionsRegistry = [];
   }
 
+  /**
+   * Add a matching function with a connection instance
+   * @private
+   * @param matchingFn - Matching function to decide what connection function to invoke
```
Comment (miralemd) [inline src/connection-matcher.js]: missing types

File: src/derived-fields-template.js [hunk 16]
```diff
@@ -1,13 +1,28 @@
...
+   * @param {callback} fieldMatchFunction - Matching function that will apply a field template definition.
```
Comment (miralemd) [inline src/derived-fields-template.js]: parameters that are properties of the `options` object need to be prefixed with `options.`, e.g.:

`@param {string} options.name - Name of derived field.`

File: src/enigma-mixin/halyard-enigma-mixin.js [hunk 17]
```diff
@@ -3,7 +3,14 @@ import convertQixGetLocalInfo from './utils';
 const CONNECTION_ERROR = 'Connection Error';
 const LOADING_ERROR = 'Loading Error';
 const SYNTAX_ERROR = 'Syntax Error';
-
+/**
+ * Create error message
+ * @private
+ * @param {string} errorType
+ * @param {objcet} qixError
```
Comment (miralemd) [inline src/enigma-mixin/halyard-enigma-mixin.js]: typo

File: src/enigma-mixin/halyard-enigma-mixin.js [hunk 18]
```diff
@@ -23,11 +30,24 @@ const halyardMixin = {
     }
   },
   extend: {
+    /**
+     * Creates a session app based on the model in the halyard instance
+     * @public
+     * @param {object} halyard - A halyard instance
```
Comment (miralemd) [inline src/enigma-mixin/halyard-enigma-mixin.js]: Should use the public `Halyard` class as type here, and everywhere an instance of `Halyard` is used

File: src/halyard.js [hunk 19]
```diff
@@ -63,16 +96,34 @@ class Halyard {
...
 
-  // Support to add hyper cube explicit or implicitly
+  /**
+   * Add hyper cube explicit or implicitly
+   * @public
+   * @param {object} arg1 - Hypercube
+   * @param {object} options - Hypercube options
+   * @param {string} name - Name
+   * @param {string} section - Section to add hypercube data to
+   * @returns {object} Hypercube
```
Comment (miralemd) [inline src/halyard.js]: use `HyperCube` as the return type

File: src/halyard.js [hunk 20]
```diff
@@ -106,6 +163,11 @@ class Halyard {
     return this.addItem(newTable);
   }
 
+  /**
+   * Verify that item doesn't exist in model
+   * @public
+   * @param {object} newItem - Table or Hypercube
```
Comment (miralemd) [inline src/halyard.js]: `Table|HyperCube` as type

**Pull request merged**